pkg/puller/puller_test.go

```diff
@@ -445,6 +447,44 @@ func TestDepthChange(t *testing.T) {
 	}
 }
 
+// TestContinueSyncing adds a single peer with PO 0 to hist and live sync only a peer
+// to test that when SyncInterval returns an error, the syncing does not terminate.
+func TestContinueSyncing(t *testing.T) {
+	t.Parallel()
+
+	var (
+		addr = test.RandomAddress()
+	)
+
+	puller, _, kad, pullsync := newPuller(opts{
+		kad: []mockk.Option{
+			mockk.WithEachPeerRevCalls(mockk.AddrTuple{Addr: addr, PO: 0}),
+			mockk.WithDepth(0),
+		},
+		pullSync: []mockps.Option{
+			mockps.WithCursors([]uint64{1}),
+			mockps.WithSyncError(errors.New("sync error"))},
```

Comment:
It would be better if this is a function which returns error for few times and then returns success. We should also test if this can restart correctly.

Reply:
>> test if this can restart correctly.
what do you mean ?

Reply:
This will always return error from SyncInterval. Instead, we should return error and then on the next call we should succeed.